data_science_pipeline/peerscout/api/main.py

```diff
@@ -75,13 +115,34 @@ def _peerscout_api():
 
         abstract = data['abstract']
         extracted_keywords = list(keyword_extractor.iter_extract_keywords(text_list=[abstract]))
-        author_suggestion = data['author_suggestion']
-        suggested_reviewing_editor_person_ids = author_suggestion['include_reviewing_editors_id']
-        suggested_senior_editor_person_ids = author_suggestion['include_senior_editors_id']
+
+        recomended_senior_editors = get_editor_recomendations_for_api(
+            senior_editor_model_dict,
+            extracted_keywords,
+            DEFAULT_N_FOR_TOP_N_EDITORS
+        )
+
+        recomended_reviewing_editors = get_editor_recomendations_for_api(
+            reviewing_editor_model_dict,
+            extracted_keywords,
+            DEFAULT_N_FOR_TOP_N_EDITORS
+        )
+
+        suggested_senior_editor_person_ids = recomended_senior_editors['person_id']
+        suggested_reviewing_editor_person_ids = recomended_reviewing_editors['person_id']
+
+        suggested_senior_editor_names = recomended_senior_editors['name']
+        suggested_reviewing_editor_names = recomended_reviewing_editors['name']
+
+        # author_suggestion = data['author_suggestion']
+        # suggested_reviewing_editor_person_ids = author_suggestion['include_reviewing_editors_id']
+        # suggested_senior_editor_person_ids = author_suggestion['include_senior_editors_id']
 
         return jsonify(get_response_json(
             senior_editor_person_ids=suggested_senior_editor_person_ids if abstract else [],
             reviewing_editor_person_ids=suggested_reviewing_editor_person_ids if abstract else [],
+            senior_editor_names=suggested_senior_editor_names if abstract else [],
```

Comment:
I guess now we don't need the condition relating to the abstract anymore - if the abstract is empty, there won't be any keywords, and therefore there shouldn't be any recommended editors.